Regression: failure to validate CBOR in zero-occurrence case (JSON works fine)

Consider this schema:

```
response = {
    "success": bool,
    "data": {* label: [* tstr]}
}
label = tstr
```

You can have zero labels, which validates fine as JSON:

```json
{"success": true, "data": {}}
```

Or one label, also validates fine as JSON:

```json
{"success": true, "data": {"hello": ["world"]}}
```

When generating CBOR with Python's `cbor2` for equivalent CBOR serializations, `one.cbor` validates just fine. `zero.cbor` fails to validate, however:

```
[ERROR] Validation of "zero.cbor" failed: error validating at cbor location /"data": map requires entry key of type label
```

CBOR files are in following zip:

[cbor.zip](https://github.com/anweiss/cddl/files/9787665/cbor.zip)

--- a/src/validator/cbor.rs
+++ b/src/validator/cbor.rs
@@ -2797,7 +2797,9 @@ where
                   .push(k.clone());
                 self.object_value = Some(v.clone());
                 let _ = write!(self.cbor_location, "/{:?}", v);
-              } else {
+              } else if (!matches!(occur, Occur::ZeroOrMore(_)) && m.is_empty())
+                || (matches!(occur, Occur::ZeroOrMore(_)) && !m.is_empty())
+              {
                 self.add_error(format!("map requires entry key of type {}", ident));
               }
 
@@ -2812,7 +2814,9 @@ where
                   .push(k.clone());
                 self.object_value = Some(v.clone());
                 let _ = write!(self.cbor_location, "/{:?}", v);
-              } else {
+              } else if (!matches!(occur, Occur::ZeroOrMore(_)) && m.is_empty())
+                || (matches!(occur, Occur::ZeroOrMore(_)) && !m.is_empty())
+              {
                 self.add_error(format!("map requires entry key of type {}", ident));
               }
               return Ok(());
@@ -2826,7 +2830,9 @@ where
                   .push(k.clone());
                 self.object_value = Some(v.clone());
                 let _ = write!(self.cbor_location, "/{:?}", v);
-              } else {
+              } else if (!matches!(occur, Occur::ZeroOrMore(_)) && m.is_empty())
+                || (matches!(occur, Occur::ZeroOrMore(_)) && !m.is_empty())
+              {
                 self.add_error(format!("map requires entry key of type {}", ident));
               }
               return Ok(());
@@ -2840,7 +2846,9 @@ where
                   .push(k.clone());
                 self.object_value = Some(v.clone());
                 let _ = write!(self.cbor_location, "/{:?}", v);
-              } else {
+              } else if (!matches!(occur, Occur::ZeroOrMore(_)) && m.is_empty())
+                || (matches!(occur, Occur::ZeroOrMore(_)) && !m.is_empty())
+              {
                 self.add_error(format!("map requires entry key of type {}", ident));
               }
               return Ok(());
@@ -2854,7 +2862,9 @@ where
                   .push(k.clone());
                 self.object_value = Some(v.clone());
                 let _ = write!(self.cbor_location, "/{:?}", v);
-              } else {
+              } else if (!matches!(occur, Occur::ZeroOrMore(_)) && m.is_empty())
+                || (matches!(occur, Occur::ZeroOrMore(_)) && !m.is_empty())
+              {
                 self.add_error(format!("map requires entry key of type {}", ident));
               }
               return Ok(());
@@ -2868,7 +2878,9 @@ where
                   .push(k.clone());
                 self.object_value = Some(v.clone());
                 let _ = write!(self.cbor_location, "/{:?}", v);
-              } else {
+              } else if (!matches!(occur, Occur::ZeroOrMore(_)) && m.is_empty())
+                || (matches!(occur, Occur::ZeroOrMore(_)) && !m.is_empty())
+              {
                 self.add_error(format!("map requires entry key of type {}", ident));
               }
               return Ok(());

